### PR TITLE
Add "Show on the primary Display" and "Show on the current Display" preferences

### PR DESCRIPTION
**Pull request by petre2dor**

### CODE REVIEW

**Comment by petre2dor** [comment]
@diegodario88 sorry for adding a bunch of commits in this PR. I did not know that this will happen when I pushed to my main.
As you can see, this PR is work in progress and it needs some cleanup.
It works quite nicely, with the exception of some weird animations that makes it look like the terminal is being move from another monitor. I will look into that whenever I get a bit of time.

Cheers!

**Comment by petre2dor** [comment]
@diegodario88 this PR is ready for review and testing now

File: quake-terminal@diegodario88.github.io/quake-mode.js [hunk 1]
```diff
@@ -343,15 +343,20 @@ export const QuakeMode = class {
...
-		const maxNumberOfMonitors = global.display.get_n_monitors() - 1;
-
-		if (mainMonitorScreen > maxNumberOfMonitors) {
-			mainMonitorScreen = maxNumberOfMonitors;
+		// get the monitor index on which the terminal will be rendered
+		const monitorIndex = this._settings.get_boolean("render-on-current-monitor")
+			? global.display.get_current_monitor()
+			: this._settings.get_boolean("render-on-primary-monitor")
+				? Main.layoutManager.primaryIndex
+				: this._settings.get_int("monitor-screen");
```
Comment (diegodario88) [inline quake-terminal@diegodario88.github.io/quake-mode.js:351]: I would suggest avoiding nested ternary operators if possible, as they add complexity to the code and make it harder to read.

**Review by diegodario88 — CHANGES_REQUESTED** [review]

File: quake-terminal@diegodario88.github.io/quake-mode.js [hunk 2]
```diff
@@ -343,15 +343,20 @@ export const QuakeMode = class {
 			return;
 		}
 
-		let mainMonitorScreen = this._settings.get_int("monitor-screen");
-		const maxNumberOfMonitors = global.display.get_n_monitors() - 1;
-
-		if (mainMonitorScreen > maxNumberOfMonitors) {
-			mainMonitorScreen = maxNumberOfMonitors;
+		// get the monitor index on which the terminal will be rendered
+		const monitorIndex = this._settings.get_boolean("render-on-current-monitor")
```
[the remark this answers is not in the document]
Comment (petre2dor) [reply quake-terminal@diegodario88.github.io/quake-mode.js]: done

File: quake-terminal@diegodario88.github.io/quake-mode.js [hunk 3]
```diff
@@ -98,6 +98,28 @@ export const QuakeMode = class {
 		return actor;
 	}
 
+	get monitorDisplayScreenIndex() {
+		if (this._settings.get_boolean("render-on-current-monitor")) {
+			// current display
```
Comment (petre2dor) [inline quake-terminal@diegodario88.github.io/quake-mode.js:103]: @diegodario88 should I drop these little comments? overkill?

Comment (diegodario88) [reply quake-terminal@diegodario88.github.io/quake-mode.js:103]: Sure thing! Yes, you can drop those little comments.

Comment (petre2dor) [reply quake-terminal@diegodario88.github.io/quake-mode.js]: done

**Comment by Henry78** [comment]
LGTM (https://github.com/diegodario88/quake-terminal/issues/15#issuecomment-1803408217)

**Pull request merged**